packages/next-swc/crates/next-core/src/next_shared/transforms/next_react_server_components.rs

```diff
@@ -0,0 +1,62 @@
+use std::path::PathBuf;
+
+use anyhow::Result;
+use async_trait::async_trait;
+use next_transform_react_server_components::{server_components, Config, Options};
+use swc_core::{
+    common::{util::take::Take, FileName},
+    ecma::{
+        ast::{Module, Program},
+        visit::FoldWith,
+    },
+};
+use turbo_tasks::Vc;
+use turbopack_binding::turbopack::{
+    ecmascript::{CustomTransformer, EcmascriptInputTransform, TransformContext},
+    turbopack::module_options::{ModuleRule, ModuleRuleEffect},
+};
+
+use super::module_rule_match_js_no_url;
+
+/// Returns a rule which applies the Next.js react server components transform.
+pub fn get_next_react_server_components_transform_rule(
+    is_react_server_layer: bool,
```

Comment:
Where is this called with `false`?

Reply:
There are some remaining failing test cases, probably need to setup this rule with this flag. Currently next-swc have 3 conditions for this transform

1. does not apply
2. false
3. true

and when I tried to put it in other places, there are failures in between 1/2 and I need to dig further to confirm when it should apply or when it should be a false.